### PR TITLE
Move toolkit bundling to a sub project

### PR DESCRIPTION
**Pull request by abrooksv**
Do not bundle the 3 "sub-plugins" into one in the root project, use a dedicated sub project for that

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-ultimate/build.gradle.kts [hunk 1]
```diff
@@ -18,12 +18,12 @@ dependencies {
...
-    val parentIntellijTask = rootProject.intellij
+    pluginName = "aws-toolkit-jetbrains"
+
     version = ideProfile.ultimate.sdkVersion
     setPlugins(*ideProfile.ultimate.plugins)
-    pluginName = parentIntellijTask.pluginName
-    updateSinceUntilBuild = parentIntellijTask.updateSinceUntilBuild
-    downloadSources = parentIntellijTask.downloadSources
+
+    downloadSources = false
```
Comment (hunterwerlla) [inline jetbrains-ultimate/build.gradle.kts:26]: Add the same comment from rider?

File: build.gradle.kts [hunk 2]
```diff
@@ -297,41 +292,15 @@ subprojects {
...
-tasks.getByName<PublishTask>("publishPlugin") {
-    token(publishToken)
-    channels(publishChannel.split(",").map { it.trim() })
-}
-
 tasks.register<GenerateGithubChangeLog>("generateChangeLog") {
     changeLogFile.set(project.file("CHANGELOG.md"))
 }
 
+val ktlint: Configuration by configurations.creating
```
Comment (hunterwerlla) [inline build.gradle.kts:303]: What does this change do?

Comment (abrooksv) [reply build.gradle.kts:303]: I just moved it next to the ktlint task

File: buildspec/linuxTests.yml [hunk 3]
```diff
@@ -42,7 +42,7 @@ phases:
       - rsync -rmq --include='*/' --include '**/build/idea-sandbox/system*/log/**' --exclude='*' . $TEST_ARTIFACTS/ || true
       - rsync -rmq --include='*/' --include '**/build/reports/**' --exclude='*' . $TEST_ARTIFACTS/ || true
       - rsync -rmq --include='*/' --include '**/test-results/**/*.xml' --exclude='*' . $TEST_ARTIFACTS/test-reports || true
-      - cp -r ./build/distributions/*.zip $BUILD_ARTIFACTS/ || true
+      - cp -r ./intellij/build/distributions/*.zip $BUILD_ARTIFACTS/ || true
```
[the remark this answers is not in the document]
Comment (abrooksv) [reply buildspec/linuxTests.yml:45]: we dont upload the plugin to the report in those specs

File: build.gradle.kts [hunk 4]
```diff
@@ -374,33 +343,16 @@ subprojects.forEach {
...
-            allTasks.forEach { it ->
-                if (it.name == "runIde" && it.project != project(":jetbrains-core")) {
-                    it.enabled = false
-                }
-            }
-        }
-    }
+tasks.named("check") {
+    dependsOn(ktlintTask)
+    dependsOn(coverageReport)
```
Comment (rli) [inline build.gradle.kts:348]: what happened to `verifyPlugin`?

File: jetbrains-ultimate/build.gradle.kts (hunk 1 above)
Comment (abrooksv) [reply jetbrains-ultimate/build.gradle.kts:26]: Re-wrote the comments, im not sure the RD one is even correct anymore (not that it matters)

File: build.gradle.kts (hunk 4 above)
Comment (abrooksv) [reply build.gradle.kts:348]: Oh good call

**Comment by hunterwerlla** [comment]
Unit tests failed because ```cp: cannot stat './intellij/build/distributions/*.zip': No such file or directory```

**Review by hunterwerlla — APPROVED** [review]

**Comment by abrooksv** [comment]
> Unit tests failed because `cp: cannot stat './intellij/build/distributions/*.zip': No such file or directory`

KTLint is killing the build before that point I think... Let me fix ktlint

**Review by hunterwerlla — APPROVED** [review]

**Pull request merged**